pkg/virt-operator/resource/apply/delete.go

```diff
@@ -463,11 +463,63 @@ func DeleteAll(kv *v1.KubeVirt,
 			return nil
 		}
 	}
+	kubeVirtLabelPrefix := "kubevirt.io/"
+	err = deleteKubeVirtLabelsFromNodes(kv, clientset, kubeVirtLabelPrefix)
+	if err != nil {
+		return err
+	}
 
 	err = deleteDummyWebhookValidators(kv, clientset, stores, expectations)
 	if err != nil {
 		return err
 	}
+
+	return nil
+}
+
+func deleteKubeVirtLabelsFromNodes(kv *v1.KubeVirt, clientset kubecli.KubevirtClient, kubeVirtLabelPrefix string) error {
+	// Get the list of nodes
+	nodeList, err := clientset.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list nodes: %v", err)
+	}
+
+	for _, node := range nodeList.Items {
+		labels := node.GetLabels()
+
+		modified := false
+
+		// If labels map is nil, continue to the next node
+		if labels == nil {
+			continue
+		}
+
+		// Iterate over the label and remove kubevirtlabels
+		for labelkey := range labels {
+			if strings.HasPrefix(labelkey, kubeVirtLabelPrefix) {
+				delete(labels, labelkey)
+				modified = true
+			}
+		}
+
+		// If any kubevirt labels were removed, update the node labels using Patch
+		if modified {
+			patchBytes, err := json.Marshal(labels)
+			if err != nil {
+				return fmt.Errorf("failed to marshal labels: %v", err)
+			}
+
+			// Construct the JSON patch operation
+			ops := fmt.Sprintf(`[{ "op": "replace", "path": "/metadata/labels", "value": %s }]`, string(patchBytes))
```

Comment:
Please use the new type PatchSet introduced by https://github.com/kubevirt/kubevirt/pull/11599